models/ed/R/model2netcdf.ED2.R

```diff
@@ -1365,24 +1499,44 @@ read_S_files <- function(sfile, outdir, pfts, pecan_names = NULL, settings = NUL
   
 } # read_S_files
 
-#helper fun to extract pft numbers or match based on pft name
-get_pft_num <- function(x) {
-  pftmapping <- pftmapping #get rid of no visible binding warning
-  pft_number <- x[["ed2_pft_number"]]
-  pft_name <- x[["name"]]
-  if(!is.null(pft_number)) {
-    pft_number <- as.numeric(pft_number)
-    if (!is.finite(pft_number)) {
-      PEcAn.logger::logger.severe(
-        "ED2 PFT number present but not parseable as number. Value was ",
-        pft_number
-      )
+
+#' Extract pft numbers from settings$pfts
+#' 
+#' A helper function to extract a named vector of pft numbers from
+#' `settings$pfts`.  Will use pft numbers in `settings` if they exist, otherwise
+#' it'll match using the `pftmapping` dataset
+#'
+#' @param pfts settings$pfts
+#'
+#' @return named numeric vector
+#'
+extract_pfts <- function(pfts) {
```

Comment:
Not a required change, but this file has gotten long and unwieldy -- might be good to move some of these helper functions to either files of their own or a general file for model2netcdf.ED2 helpers.